core/src/modules/ops/record.ts

```diff
@@ -945,38 +960,112 @@ export namespace RecordOps {
     limit = 100,
     toExport = true
   ) {
-    let records: GrouparooRecord[] = await api.sequelize.query(
+    const records: GrouparooRecord[] = await api.sequelize.query(
       `
-    SELECT "id" from "records" where "state" = 'pending'
-    EXCEPT
-    SELECT DISTINCT("recordId") FROM "recordProperties" WHERE "state" = 'pending'
-    LIMIT ${limit}
-    ;
+      SELECT "id" from "records" where "state" = 'pending'
+      EXCEPT
+      SELECT DISTINCT("recordId") FROM "recordProperties" WHERE "state" = 'pending'
+      LIMIT ${limit};
     `,
       {
         type: QueryTypes.SELECT,
         model: GrouparooRecord,
+        include: RecordProperty,
       }
     );
 
-    const updateResponse = await GrouparooRecord.update(
-      { state: "ready" },
+    if (!records.length) {
+      return [];
+    }
+
+    // const [, updateResponse] = await GrouparooRecord.update(
+    //   { state: "ready" },
+    //   {
+    //     where: {
+    //       id: { [Op.in]: records.map((p) => p.id) },
+    //       state: "pending",
+    //     },
+    //   }
+    // );
+
+    const invalidRecords = await GrouparooRecord.findAll({
+      where: {
+        id: { [Op.in]: records.map((p) => p.id) },
+        invalid: false,
+      },
+      include: {
+        model: RecordProperty,
+        where: {
+          invalidReason: {
+            [Op.not]: null,
+          },
+        },
+      },
+    });
+    const [, invalidUpdateResponse] = await GrouparooRecord.update(
+      { invalid: true },
       {
         where: {
-          id: { [Op.in]: records.map((p) => p.id) },
-          state: "pending",
+          id: {
+            [Op.in]: invalidRecords.map((record) => record.id),
+          },
         },
       }
     );
```

Comment:
I bet there's a single statement you can use to update the records without selecting them